create-aleo-app/template-extension/src/background.js

```diff
@@ -0,0 +1 @@
+new Worker(new URL("worker.js", import.meta.url), { type: "module" });
```

Comment:
Is this the main worker that does all of the program execution? If an extension builder wanted to spin up another - how would that be done?

Reply:
Yes, the SDK must be run in a Worker, so this runs the SDK code in a Worker. The SDK code is in the `worker.js` file.

I don't know why an extension would want to have multiple workers running SDK code, but if they wanted to they would just create a new `worker2.js` file, add it into the Rollup config:

```js
    input: {
        background: "./src/background.js",
        worker: "./src/worker.js",
        worker2: "./src/worker2.js",
    },
```

And then spawn it in the `background.js` file:

```js
new Worker(new URL("worker.js", import.meta.url), { type: "module" });

new Worker(new URL("worker2.js", import.meta.url), { type: "module" });
```

Reply:
Aleo functions are compute heavy so they might need to run multiple workers to execute multiple functions concurrently.